barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -1006,25 +1057,31 @@ void ContentAddressedCachedTreeStore<LeafValueType>::remove_historical_block(con
                                             forkConstantData_.name_));
         }
     }
-    WriteTransactionPtr writeTx = create_write_transaction();
-    try {
-        std::optional<index_t> maxIndex = std::nullopt;
-        // remove the historical block's node data
-        remove_node(std::optional<fr>(blockData.root), 0, maxIndex, *writeTx);
-        // remove the block's entry in the block table
-        dataStore_->delete_block_data(blockNumber, *writeTx);
-        // increment the oldest historical block number as committed data
-        committedMeta.oldestHistoricBlock++;
-        persist_meta(committedMeta, *writeTx);
-        writeTx->commit();
-    } catch (std::exception& e) {
-        writeTx->try_abort();
-        throw std::runtime_error(format("Unable to commit removal of historical block: ",
-                                        blockNumber,
-                                        ". Tree name: ",
-                                        forkConstantData_.name_,
-                                        " Error: ",
-                                        e.what()));
+    {
+        WriteTransactionPtr writeTx = create_write_transaction();
+        try {
+            // If the tree was empty at the block being removed then we should not attempt to remove
+            // any nodes. (there were no nodes at the point this block was comitted)
+            if (blockData.size > 0) {
```

Comment:
And again, don't attempt to remove historical nodes if the tree was empty at the time.